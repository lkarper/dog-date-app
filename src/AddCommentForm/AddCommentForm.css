.AddCommentForm__form {
    width: 80%;
}

.AddCommentForm__textarea {
    width: 100%;
}