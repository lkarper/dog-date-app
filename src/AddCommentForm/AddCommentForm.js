import React, { useState, useContext } from 'react';
import { v4 as uuidv4 } from 'uuid';
import UserContext from '../contexts/UserContext';
import './AddCommentForm.css';

const AddCommentForm = (props) => {

    const context = useContext(UserContext);

    const { reviewId } = props;

    const [commentText, setCommentText] = useState('');

    const handleSubmit = (event) => {
        event.preventDefault();
        const newComment = {
            id: uuidv4(),
            review_id: reviewId,
            commenter: context.user.username,
            date_time: new Date().toJSON(),
            comment: commentText,
        };
        context.addComment(newComment);
        setCommentText('');
    }

    return (
        <form 
            className='AddCommentForm__form'
            onSubmit={handleSubmit}
        >
            <textarea
                className='AddCommentForm__textarea'
                aria-label='Leave your own comment here.'
                placeholder='(Leave your own comment here.)'
                maxLength='2000'
                rows='8'
                value={commentText}
                onChange={(e) => setCommentText(e.target.value)}
            >
            </textarea> 
            <button
                type='submit'
                disabled={commentText.trim().length === 0}
            >
                Submit
            </button>
        </form>
    )

}

export default AddCommentForm;