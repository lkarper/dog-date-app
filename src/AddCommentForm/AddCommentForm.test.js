import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import AddCommentForm from './AddCommentForm';

describe('AddCommentForm component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <AddCommentForm />
            </BrowserRouter>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(
            <BrowserRouter>
                <AddCommentForm />
            </BrowserRouter>
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});