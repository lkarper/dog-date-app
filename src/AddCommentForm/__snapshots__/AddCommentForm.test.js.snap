// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCommentForm component renders the UI as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <AddCommentForm
      comments={Array []}
      id=""
      oldComment=""
      reviewId=""
      setComments={[Function]}
      suffix=""
    >
      <p
        className="AddCommentForm__error error"
      >
        Cannot comment at this time.
      </p>
    </AddCommentForm>
  </Router>
</BrowserRouter>
`;
